perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_sshPublicKey.java

```diff
@@ -8,29 +9,248 @@
+import java.security.spec.DSAPublicKeySpec;
+import java.security.spec.ECParameterSpec;
+import java.security.spec.ECPoint;
+import java.security.spec.RSAPublicKeySpec;
+import java.security.spec.ECPublicKeySpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.ECGenParameterSpec;
+import java.security.spec.InvalidParameterSpecException;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * @author Jakub Peschel <jakubpeschel@gmail.com>
  */
 public class urn_perun_user_attribute_def_def_sshPublicKey extends UserAttributesModuleAbstract implements UserAttributesModuleImplApi {
 
+	private final Pattern pattern = Pattern.compile("^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$");
+	final String RSA = "ssh-rsa";
+	final String DSA = "ssh-dsa";
+	final String ECDSA_SHA2_NISTP256 = "ecdsa-sha2-nistp256";
+	final String ECDSA_SHA2_NISTP384 = "ecdsa-sha2-nistp384";
+	final String ECDSA_SHA2_NISTP521 = "ecdsa-sha2-nistp521";
+	final String ED25519 = "ssh-ed25519";
+	final String SK_ED25519 = "sk-ed25519";
+	final String SK_ECDSA = "sk-ecdsa";
+
 	@Override
 	public void checkAttributeSyntax(PerunSessionImpl perunSession, User user, Attribute attribute) throws WrongAttributeValueException {
 		//Null in value is ok here
-		if(attribute.getValue() == null) return;
-		
+		if (attribute.getValue() == null) return;
+
 		//Testing if some ssh key contains new line character
 		List<String> sshKeys = attribute.valueAsList();
-		for(String sshKey: sshKeys) {
-			if(sshKey != null) {
-				if(sshKey.contains("\n")) throw new WrongAttributeValueException(attribute, user, "One of keys in attribute contains new line character. New line character is not allowed here.");
+		for (String sshKey : sshKeys) {
+			if (sshKey != null) {
+				if (sshKey.contains("\n"))
+					throw new WrongAttributeValueException(attribute, user, "One of keys in attribute contains new line character. New line character is not allowed here.");
+				try {
+					validateSSH(sshKey);
+				} catch (Exception e) {
+					throw new WrongAttributeValueException("Invalid SSH key format: " + e.getMessage());
+				}
 			}
 		}
 	}
 
+	/**
+	 * Checks whether is the SSH key in the correct format. Throws exception if validation fails for any reason.
+	 *
+	 * @param sshKey SSH key to be checked
+	 * @exception Exception that is thrown whenever SSH key is not in correct format
+	 */
+	private void validateSSH(String sshKey) throws Exception {
+		byte[] bytes = null;
+		int[] pos = {0};
+
+		// look for the Base64 encoded part of the line to decode
+		boolean typeFound = false;
+		String sshKeyTypeValue = "";
+
+		List<String> sshTypes = Arrays.asList(RSA, DSA, ECDSA_SHA2_NISTP256, ECDSA_SHA2_NISTP384, ECDSA_SHA2_NISTP521, ED25519);
+
+		for (String part : sshKey.split(" ")) {
+			Matcher matcher = pattern.matcher(part);
+			if (matcher.matches()) {
```

Comment:
This way you do not check whether the ssh type is put directly before the bsae64 part.

Reply:
```
if (mathcer.matches()) {
...
} else if (typeFound) {
error
}
```